packages/too-many-hooks/src/useMatchMedia.ts

```diff
@@ -0,0 +1,50 @@
+import { useCallback, useEffect, useState } from 'react'
+
+/**
+ * `useMatchMedia` hook type
+ *
+ * @export
+ * @typedef {UseMatchMedia}
+ * @param {string} query - a CSS media query to match on
+ * @param {?boolean} [default=false] - default match value
+ * @returns {boolean}
+ */
+export type UseMatchMedia = (query: string, defaultValue?: boolean) => boolean
+
+/**
+ * Returns a boolean indicating whether or not a media query matches
+ *
+ * @example
+ * To check for a screen width <= 360px
+ * ```ts
+ * const isMobile = useMatchMedia("screen and (max-width: 360px)")
+ * ```
+ * @example
+ * To check if the browser is printing the page
+ * ```ts
+ * const isPrinting = useMAtchMedia("print")
+ * ```
+ * @implements {UseMatchMedia}
+ * @param {string} query - a CSS media query to match on
+ * @param {?boolean} [default=false] - default match value
+ * @returns {boolean}
+ */
+const useMatchMedia: UseMatchMedia = (query: string, defaultValue = false): boolean => {
+  const [isMediaQueryMatching, setIsMediaQueryMatching] = useState<boolean>(defaultValue)
+
+  const onMediaQueryListChange = useCallback(
+    (e: MediaQueryListEvent) => setIsMediaQueryMatching(e.matches),
+    [],
+  )
+
+  useEffect(() => {
+    const mediaQueryList = window.matchMedia(query)
+    setIsMediaQueryMatching(mediaQueryList.matches)
+    mediaQueryList.addEventListener('change', onMediaQueryListChange)
+    return () => mediaQueryList.removeEventListener('change', onMediaQueryListChange)
+  }, [query, onMediaQueryListChange])
+
+  return isMediaQueryMatching
```

Comment:
I wonder if it would be more consistent with the hook name if we returned the whole `MediaQueryListEvent` from `window.matchMedia`. What it's really doing now is more like `useMatchMediaMatches` 🤔

Reply:
I had it return the `MediaQueryList` because the event has additional fields and can only be grabbed by a change so on the first render there is no check for the matching unless using the actual list